### PR TITLE
Update course users using javascript for user management pages

### LINKED ISSUE #1252: Cannot set user to phantom?
To reproduce:
Go to students page, tick the Phantom checkbox, and click save. Nothing happened. Tested both locally and on staging server

### PR DESCRIPTION
**Pull request by kxmbrian**
Fixes #1252 by treating a table row itself as a form instead of having a form or div nested in it.

### CODE REVIEW

File: app/assets/javascripts/course/users.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//= require helpers/form_helpers
```
Comment (allenwq) [inline app/assets/javascripts/course/users.js]: Are you going to generalise this for admin user management to use ?

File: app/views/course/users/requests.html.slim [hunk 2]
```diff
@@ -18,11 +18,11 @@ div.users
...
         - @course_users.each do |user|
-          = simple_form_for [current_course, user], resource: :course_user, \
-            wrapper: :inline_form do |f|
-            = f.error_notification
-            = f.error :course_user
-            tr
+          = content_tag_for(:tr, user, 'data-action' => course_user_path(current_course, user),
+                                       'data-method' => 'patch')
+            = simple_fields_for user, resource: :course_user, wrapper: :inline_form do |f|
+              = f.error_notification
```
Comment (allenwq) [inline app/views/course/users/requests.html.slim]: Is this still necessary ?

File: spec/features/course/student_management_spec.rb [hunk 3]
```diff
@@ -20,27 +20,24 @@
       end
     end
 
-    scenario "Course staff can change students' names" do
+    scenario "Course staff can update students' records", js: true do
+      student_to_update = course_students[Random.rand(course_students.length)]
```
Comment (allenwq) [inline spec/features/course/student_management_spec.rb]: `course_students.sample` will do.

File: app/assets/javascripts/course/users.js (hunk 1 above)
Comment (kxmbrian) [inline app/assets/javascripts/course/users.js]: Yes, will see how to reuse this when I work on that PR.

**Pull request merged**